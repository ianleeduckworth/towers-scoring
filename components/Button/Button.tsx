import * as React from 'react';
import {
  Text, TouchableOpacity, View, StyleSheet,
} from 'react-native';
import { defaults, globalStyles } from '../../styles/global';

interface ButtonProps {
    text: string;
    onPress: () => void;
}
export const Button = ({
  text,
  onPress,
}: ButtonProps) => (
        <TouchableOpacity onPress={onPress} style={styles.buttonContainer}>
            <View style={styles.button}>
                <Text style={styles.buttonText}>{text}</Text>
            </View>
        </TouchableOpacity>
);

const styles = StyleSheet.create({
  button: {
    backgroundColor: defaults.mainColor,
    borderRadius: 5,
    paddingHorizontal: 10,
    paddingVertical: 14,
  },
  buttonContainer: {
    minWidth: 360,
  },
  buttonText: {
    ...globalStyles.paragraphText,
    color: defaults.white,
    textAlign: 'center',
    textTransform: 'uppercase',
  },
});
