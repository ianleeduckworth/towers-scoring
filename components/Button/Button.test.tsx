import renderer from 'react-test-renderer';
import * as React from 'react';
import { Button } from './Button';
import { noOp } from '../../utilities/test/testUtilities';

describe('button', () => {
  it('renders correctly', () => {
    const element = renderer.create(
            <Button text="Test Button" onPress={noOp} />,
    ).toJSON();

    expect(element).toMatchSnapshot();
  });
});
