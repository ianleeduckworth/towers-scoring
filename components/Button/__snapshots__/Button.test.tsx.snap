// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`button renders correctly 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "minWidth": 360,
      "opacity": 1,
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "#476A6F",
        "borderRadius": 5,
        "paddingHorizontal": 10,
        "paddingVertical": 14,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#FFFFFF",
          "fontFamily": "Roboto",
          "fontSize": 18,
          "fontWeight": "bold",
          "textAlign": "center",
          "textTransform": "uppercase",
        }
      }
    >
      Test Button
    </Text>
  </View>
</View>
`;
